src/Workspaces/Core/Portable/CodeGeneration/CodeGenerationHelpers.cs

```diff
@@ -102,6 +103,42 @@ public static bool IsSpecialType(ITypeSymbol type, SpecialType specialType)
             return type != null && type.SpecialType == specialType;
         }
 
+        public static bool IsFlowAnalysisAttribute(AttributeData attribute)
+        {
+            return GetNameParts(attribute.AttributeClass.ContainingNamespace).SequenceEqual(new[] { "System", "Diagnostics", "CodeAnalysis" }) &&
+                attribute.AttributeClass.Name switch
+                {
+                    nameof(AllowNullAttribute) => true,
+                    nameof(DisallowNullAttribute) => true,
+                    nameof(MaybeNullAttribute) => true,
+                    nameof(NotNullAttribute) => true,
+                    nameof(MaybeNullWhenAttribute) => true,
+                    nameof(NotNullWhenAttribute) => true,
+                    nameof(NotNullIfNotNullAttribute) => true,
+                    nameof(DoesNotReturnAttribute) => true,
+                    nameof(DoesNotReturnIfAttribute) => true,
```

Comment:
are there not SpecialTypes for these?  @jcouv ?

Reply:
We don't have WellKnownTypes for those. For attributes we usually have an `AttributeDescription` (which describes name and constructor signature). But I don't think those are part of public compiler API at the moment.

Reply:
Can do this by getting the actual type from the compilation and checking against that.

Reply:
How can I get the `Compilation` from within the generator?